src/storage/data_object_type_registry.rs

```diff
@@ -0,0 +1,154 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use parity_codec::Codec;
+use parity_codec_derive::{Encode, Decode};
+use srml_support::{StorageMap, StorageValue, decl_module, decl_storage, decl_event, ensure, Parameter};
+use runtime_primitives::traits::{SimpleArithmetic, As, Member, MaybeSerializeDebug, MaybeDebug};
+use system::{self, ensure_root};
+use crate::traits;
+
+pub trait Trait: system::Trait + MaybeDebug
+{
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    type DataObjectTypeID: Parameter + Member + SimpleArithmetic + Codec + Default + Copy
```

Comment:
By following Substrate and our naming convention for IDs, it should be: `ID` -> `Id`
```suggestion
    type DataObjectTypeId: Parameter + Member + SimpleArithmetic + Codec + Default + Copy
```

Reply:
Yeah, that's a naming convention that I've seen a lot in Java, and it's pretty stupid. ID is not a word, it's an abbreviation, so CamelCasing it makes no sense at all. Still, I will change it.

Reply:
Here is a quote from [ID article](https://en.wikipedia.org/wiki/ID) on Wiki:

> ID, Id, id or I.D. may refer to:
> 
> * Identity document, any document used to verify a person's identity
> * Identifier, a symbol which uniquely identifies an object or record

I believe that in our cases (`AccounId`, `MemberId`, `ProposalId`) an `Id` stands for `Identifier` (i.e. **id**_entifier. ID could be considered as an abbreviation only in case if we reffer to **I**_dentity **D**_ocument – that is not our case at all.